### PR TITLE
Update profile user

### PR DESCRIPTION
**Pull request by ducminhvu**

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -18,9 +21,31 @@ def create
...
+    if @user.update_attributes user_params
+      flash[:success] = t :update_success
+      redirect_to @user
+    else
+      render "edit"
+    end
+  end
+
   private
+
```
Comment (phamkykhoi) [inline app/controllers/users_controller.rb]: Remove line empty

File: app/models/user.rb [hunk 2]
```diff
@@ -42,4 +45,11 @@ def authenticated? attribute, token
     return false if digest.nil?
     BCrypt::Password.new(digest).is_password? token
   end
+
+  # Validates the size of an uploaded picture.
+  def picture_size
+    if picture.size > 5.megabytes
```
Comment (phamkykhoi) [inline app/models/user.rb]: Settings

File: app/views/users/edit.html.erb [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      <%= f.label :password %>
+      <%= f.password_field :password, class: "form-control" %>
+
+      <%= f.label :password_confirmation, t("confirm_label") %>
+      <%= f.password_field :password_confirmation, class: "form-control" %>
+
+      <span class="picture">
+        <%= f.label :picture, t("update_avatar") %><br />
+        <%= image_tag @user.picture.url if @user.picture? %>
+        <%= f.file_field :picture, accept: "image/jpeg,image/gif,image/png" %>
```
Comment (phamkykhoi) [inline app/views/users/edit.html.erb]: Settings

File: app/views/users/edit.html.erb [hunk 4]
```diff
@@ -0,0 +1,30 @@
+<% provide(:title, t("edit_profile")) %>
+<h1><%= t("edit_profile") %></h1>
+
+<div class="row">
+  <div class="col-md-6 col-md-offset-3">
+    <%= form_for(@user) do |f| %>
```
Comment (phamkykhoi) [inline app/views/users/edit.html.erb]: Remove ( )

File: app/controllers/users_controller.rb [hunk 5]
```diff
@@ -18,9 +21,31 @@ def create
...
   def user_params
     params.require(:user).permit :username, :email, :password,
-      :password_confirmation
+      :password_confirmation, :picture
+  end
+
+  def correct_user
+    if current_user.id.to_s != params[:id]
```
Comment (khanhkhanh) [inline app/controllers/users_controller.rb]: .to_s is needed? If not, remove it.

File: app/models/user.rb (hunk 2 above)
Comment (khanhkhanh) [inline app/models/user.rb]: 5 => Settings

File: app/views/users/edit.html.erb [hunk 6]
```diff
@@ -0,0 +1,30 @@
+<% provide(:title, t("edit_profile")) %>
+<h1><%= t("edit_profile") %></h1>
```
Comment (khanhkhanh) [inline app/views/users/edit.html.erb]: Remove ()

File: app/views/users/edit.html.erb [hunk 7]
```diff
@@ -0,0 +1,30 @@
+<% provide(:title, t("edit_profile")) %>
```
Comment (khanhkhanh) [inline app/views/users/edit.html.erb]: Remove ()

File: app/views/users/edit.html.erb (hunk 4 above)
Comment (khanhkhanh) [inline app/views/users/edit.html.erb]: Remove ()

File: config/settings.yml [hunk 8]
```diff
@@ -3,3 +3,4 @@ CHECKBOOK_TRUE: "1"
 email_maximum: 255
 name_maximum: 50
 password_minimum: 6
+size_picture: 50
```
Comment (khanhkhanh) [inline config/settings.yml]: Break line

File: app/controllers/users_controller.rb (hunk 5 above)
Comment (ducminhvu) [inline app/controllers/users_controller.rb]: Yes, Current_user.id is int and params[:id] is a string

File: app/views/users/edit.html.erb (hunk 6 above)
Comment (ducminhvu) [inline app/views/users/edit.html.erb]: You don't know?? this () can't remove

File: app/assets/javascripts/application.js [hunk 9]
```diff
@@ -15,3 +15,12 @@
 //= require bootstrap
 //= require turbolinks
 //= require_tree .
+
+<script type="text/javascript">
```
Comment (phamkykhoi) [inline app/assets/javascripts/application.js]: In file .js not use <script type="text/javascript">

File: app/models/user.rb [hunk 10]
```diff
@@ -42,4 +45,11 @@ def authenticated? attribute, token
     return false if digest.nil?
     BCrypt::Password.new(digest).is_password? token
   end
+
+  # Validates the size of an uploaded picture.
```
Comment (phamkykhoi) [inline app/models/user.rb]: Remove comment

File: app/controllers/users_controller.rb [hunk 11]
```diff
@@ -18,9 +21,30 @@ def create
...
+
   private
   def user_params
     params.require(:user).permit :username, :email, :password,
-      :password_confirmation
+      :password_confirmation, :picture
+  end
+
+  def correct_user
+    if current_user.id.to_s != params[:id]
```
Comment (dieunb) [inline app/controllers/users_controller.rb]: unless

**Comment by phamkykhoi** [comment]
OK

**Pull request merged**